stringbasic/String.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+
+class String {
+public:
+
+	String();
+
+	String(const char* string);
+
+	String(const String&);
+
+	String& operator=(const String&);
```

Comment:
I'm glad you've taken care of copy assignment/CTOR, but what about move assignment/CTOR?
If you want the default behavior - I still think it's better to use "= default" to let devs know you've done this intentionally.